dashmips/syscalls/print_syscalls.py

```diff
@@ -1,42 +1,56 @@
 """Syscalls related to printing."""
 from . import mips_syscall
 from ..models import MipsProgram
-from ..utils import intify
+from ..utils import intify, bytesify
+from struct import Struct
 
 
 @mips_syscall(4)
 def print_string(program: MipsProgram):
-    """Print string at address provided in $a0."""
+    """Print string. $a0 = address of null-terminated string to print."""
     address = program.registers["$a0"]
     string = program.memory.read_str(address)
-    print(string, end="")
+    print(string, end=" ")
 
 
 @mips_syscall(11)
 def print_char(program: MipsProgram):
-    """Print string at address provided in $a0."""
+    """Print character. $a0 = character to print."""
     character = chr(program.registers["$a0"] & 0xFF)
     if ord(" ") <= ord(character) <= ord("~") or character in ("\n", "\t", "\r"):
         # is printable
-        print(character, end="")
+        print(character, end=" ")
 
 
 @mips_syscall(1)
 def print_int(program: MipsProgram):
-    """Print Int in Dec."""
-    print(program.registers["$a0"], end="")
+    """Print Int in Dec. $a0 = integer to print."""
+    print(program.registers["$a0"], end=" ")
```

Comment:
Whats the goal behind printing with trailing space as opposed to nothing?

Reply:
That was just to match the output Jen had in Homework 1, but it's more of a design decision than anything else.  I figured we should ask her what she'd prefer and can always change it back!
![Screen Shot 2020-07-23 at 9 38 33 AM](https://user-images.githubusercontent.com/44068487/88313440-6815b500-ccc8-11ea-8229-b27f4e4ba587.png)

Reply:
I think the print syscall should do only what it says it does, so just print a binary number for example, someone doing this homework would have to print the space, And for that matter they're going to have to print the new line to get the next output to appear that way.

Reply:
Alright!  I removed all the spaces from the print calls.